ansys/mapdl/reader/archive.py

```diff
@@ -251,18 +263,28 @@ def quality(self):
             )
         return quality(self._grid)
 
-    @wraps(pv.plot)
-    def plot(self, *args, **kwargs):
-        """Plot the mesh"""
-        if self._grid is None:  # pragma: no cover
-            raise AttributeError(
-                "Archive must be parsed as a vtk grid.\n" "Set `parse_vtk=True`"
-            )
-        kwargs.setdefault("color", "w")
-        kwargs.setdefault("show_edges", True)
-        self.grid.plot(*args, **kwargs)
+    if are_graphics_available:
+
+        @graphics_required
+        @wraps(pv.plot)
+        def plot(self, *args, **kwargs):
+            """Plot the mesh"""
+            if self._grid is None:  # pragma: no cover
+                raise AttributeError(
+                    "Archive must be parsed as a vtk grid.\n" "Set `parse_vtk=True`"
+                )
+            kwargs.setdefault("color", "w")
+            kwargs.setdefault("show_edges", True)
+            self.grid.plot(*args, **kwargs)
+
+    else:
+
+        def plot(self, *args, **kwargs):
+            """Placeholder for plot when graphics dependencies are unavailable."""
+            raise ImportError(ERROR_GRAPHICS_REQUIRED)
```

Comment:
Feel free to share what you think about this workaround and if another methods would be more suitable

Reply:
This should probably be a decorator around `plot` function.... actually what is `graphics_required` for then?

Reply:
``graphics_required`` is a new [decorator](https://github.com/ansys/pymapdl-reader/blob/0b30e1233909a76b305621d2f8f7a5b4032bd5d5/ansys/mapdl/reader/misc/checks.py#L55) introduced in this PR.

However, when ``graphics`` dependencies are not installed, an error is raised when loading the file even though this decorator exists. I tried to create a placeholder that would be defined when ``pyvista`` is not installed but the error persists.

Reply:
I would "fake" the import of pyvista:

```py
try:
    import pyvista as pv
except:
    pv = None

...

@graphics_required
@wraps(None if pv else pv.plot)
...
```

Reply:
That way you can get rid of the ugly duplication.

Reply:
I like this approach better than the one I implemented, thank you very much!